satellite/metainfo/service_test.go

```diff
@@ -55,3 +57,74 @@ func TestIterate(t *testing.T) {
 		require.Equal(t, 1, itemCount)
 	})
 }
+
+func TestUpdatePiecesDuplicateNodeID(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 6, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		saPeer := planet.Satellites[0]
+		uplinkPeer := planet.Uplinks[0]
+
+		// create test buckets
+		err := uplinkPeer.CreateBucket(ctx, saPeer, "test1")
+		require.NoError(t, err)
+
+		// upload an object
+		expectedData := testrand.Bytes(50 * memory.KiB)
+		err = uplinkPeer.Upload(ctx, saPeer, "test1", "test/path", expectedData)
+		require.NoError(t, err)
+
+		pointer, path := getRemoteSegment(t, ctx, saPeer)
+
+		duplicatedNodeID := storj.NodeID{}
+		pieceToRemove := make([]*pb.RemotePiece, 1)
+		pieceToAdd := make([]*pb.RemotePiece, 1)
+		pieces := pointer.GetRemote().GetRemotePieces()
+		for _, piece := range pieces {
+			if pieceToRemove[0] == nil {
+				pieceToRemove[0] = piece
+				continue
+			}
+
+			if pieceToRemove[0].NodeId != piece.NodeId {
```

Comment:
if we assume there are no duplicate node IDs to begin with, this check shouldn't be necessary. But it does make it clear what we are trying to do.

Reply:
That's kinda my intention